### DISCUSSION TITLE
Micronaut 3 - chunked transfer encoding when returning HttpResponse<Void> from an endpoint

### DISCUSSION THREAD

**Discussion by mrwilby** [micronaut-projects/micronaut-core#10124] [Q&A]
We're using Micronaut 3.x.

We've observed that if we return HttpResponse<Void> from a controller endpoint, micronaut/netty returns transfer-encoding: chunked to HTTP 1.1 clients.

```
    @Post("/")
    public HttpResponse<Void> endpointMethod(...) {
...
}

```

If we change our endpoint to explicitly return (java) void, then we do NOT see the chunked encoding response and instead, the content length response header is explicitly set to 0.

Possibly this relates to this logic:

https://github.com/micronaut-projects/micronaut-core/blob/ecc138741de6f7eef411cbbe0fdb1ed96109571f/http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java#L498

In the reactive response docs, https://docs.micronaut.io/latest/guide/#reactiveResponses, there is this note:

```When returning a Reactive type, its type affects the returned response. For example, when returning a [Flux](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html), the Micronaut framework cannot know the size of the response, so Transfer-Encoding type of Chunked is used. Whilst for types that emit a single result such as [Mono](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html) the Content-Length header is populated.```

If the controller returns HttpResponse<Void> is chunked transfer encoding the expected behavior ?

**Comment by yawkat** [comment]
is this still a thing in 4? sounds like a bug.
